A handle

Neon22 opened the pull request:
Simple handle for carriage.
Goes over existing holes. Does not foul the tension adjusters.
Uses existing holes. Need longer screws

Neon22 committed:
A handle

Simple handle for carriage.
Goes over existing holes. Does not foul the tension adjusters

Grumskiz commented:
I've printed it today and it fits perfectly. I think this is a great effort, thank you very much!
As of right now I was not able to test it in prolonged/actual knitting use yet.

There are two problems I faced though:
- in the code, you use the variable `cylres30` which is never defined
- the screw holes along the sloped edges should be countersunk, so that the screws are flat against the plastic. I used a drill bit to achieve this, but I think it's better to do it in the design itself

For anyone who wants to try:
The screw in the middle needs to be replaced with one about 6mm longer than before.
For the other two screws it's difficult to say, because right now it depends on how far you countersink the hole.

Neon22 commented:
OK. The handle is actually built on top of the scad_enhancement branch. This added "proper" screws. (and `cylres30 = 30;`)
That central hole is also supposed to be wider and so you would not need a longer screw. The middle screw is essentially not used to hold the handle down (very difficult to screw in in that position)

Alas I just checked and I have screwed up somewhere. There is indeed no recess visible.
I will fix it when/if that branch gets merged.
I'm glad you were able to force a solution.

BTW that branch is here: https://github.com/Neon22/KnittingMachine/tree/scad_enhancement

Grumskiz commented:
Oops my bad, maybe I should have paid more attention to your code instead of just forcing the handle to show up :sweat_smile:  

I managed to do fine on the middle screw with an angled wrench, but I think your intended solution is superior!

ScarlettSparks commented:
I'm finally back to my usual workspace/tools/schedule (yay!). I have quite a lot of catching up to do but plan to dive into giving these PRs a proper review later this week, thanks so much for your patience.

ScarlettSparks commented:
> The handle is actually built on top of the scad_enhancement branch.

Since this is based on the same branch as #13, I'll hold off on reviewing anything here until that PR is resolved. Thanks!

ScarlettSparks commented:
OK, I've closed out the other two PRs based on the `scad_enhancement` branch, but I think this looks isolated enough that I can review it independently. @Grumskiz have you had a chance to knit with it since your previous comment?

Neon22 commented:
Hey Scarlett, would you like me to rebuild and resubmit any of the screws stuff ? If so should I build on main or wait a bit for you to get your changes in.

This handle specifically needs an adjustment for the mounting holes

Neon22 committed:
corrected for merging to main

central hole is large so does not foul existing screw

side holes have insets for head to go inside handle body

Neon22 commented:
added inset holes and wide hole in middle to avoid crowding existing screw head

ScarlettSparks commented:
> Hey Scarlett, would you like me to rebuild and resubmit any of the screws stuff ? If so should I build on main or wait a bit for you to get your changes in.

Let's hold off on the screws rewrite for now, but thank you for offering. My time has been very scattered lately and I don't want to ask you to resubmit anything until I'm confident I can commit to reviewing it in a reasonable time frame. 😅 

Thanks for making the changes here to compare against the `main` branch. I'll be back to review shortly!

ScarlettSparks commented on SCAD/parts/handle-01.scad:
```diff
@@ -0,0 +1,101 @@
+
+
+Delta = 0.1;  // tiny adj for neater boolean subtractions
+cylres30 = 30;
```
I'd prefer to name variables like this according to their purpose, rather than their assigned numerical value; I think it avoids confusion if that value ever needs to change in the future, and otherwise I don't see any advantage over just using the literal `$fn = 30`. Since it looks like you're using the same value for all the cylinders in this file, we could keep it simple for now and call it `cylRes`? Or if you wanted to differentiate the screw resolution from the handle curves, we could do something like `screwRes` and `handleCurveRes`.

Neon22 replied:
I guess I'd prefer to have the value set in `params.scad` (same for Delta).
I put it in here to avoid changing so many files and I wasn't sure if you wanted me to proceed.

How about I amend PR and:
- put Delta in params
- add something like `resSimple = 30;` into params also as well.

or leave it in here for the time being and you can refactor into params at a later date if there's more of them.

The naming convention is tricky. Reasons why I make lower and higher res settings are:
 - bolts, or holes often are small in diameter and 3D printers are inaccurate at this res. I have found a six sided hole for screws etc works well at the 2-5mm diameter range.
 - A large diameter circle like a cam, would need often >360 resolution.
 - hull operations and some other booleans really slow down if the faces are large numbers so a simple default like 720 can make it very slow to create an object.
So I have found it useful to have a naming convention for cylinder (or sphere) resolutions.

some choices in your style might be:
`resSomedescriptor`, `facecountSomedescriptor`, 'fcSomedescriptor`, ...
where Somedescriptor might be `Bolts`, `MidRes`, `HiRes`, ...
happy to follow your requirements.

BTW I use `Delta` quite a bit so the drawn object on screen doesn't have flickering faces when everything perfectly aligns.
I use 0.1 so I can visibly see the difference when setting it up but 0.01 also works very well. 0.1 of a mm is, in my experience, below the printing res of objects for the purposes of clearance etc, and therefore does not affect the final physical object in terms of tolerance.

ScarlettSparks replied:
> Reasons why I make lower and higher res settings are:

For sure, I have no argument against setting lower `$fn` values strategically for smaller shapes; I set a relatively high project-wide default facet number in `params` to accommodate larger radii (where a too-low value might be costly in materials/print time if overlooked) but I agree that it's overkill for a 3-4mm screw hole. I'll consider these points as part of future refactoring but that's beyond the scope of this PR. 

The specific change I am requesting here is a more resilient variable name. `resSimple` sounds fine to me. 👍 

> or leave it in here for the time being and you can refactor into params at a later date if there's more of them.

Yes, please keep `resSimple` and `Delta` in this file for now. 👍 I will decide whether to incorporate them into the main params later.

Neon22 replied:
ok. submitted

ScarlettSparks commented on SCAD/parts/handle-01.scad:
```diff
@@ -0,0 +1,101 @@
+module handle_screws() {
+	width = SPONGE_BAR + 8;
+	carriageScrews();
+	translate([CAM_PLATE_WIDTH-10.8,-109,28])
```
Could you clarify where these translation values came from? Do they line up with the existing carriage screw holes?

Neon22 replied:
Yeah this is a mess, I manually lined them up with the existing` carriage screw` object (which is my master ref for this).
The work done previously on screws obviated these kinds of calcs but maybe at a later date the values would be amended so no magic values were required.

Otherwise I could go back to the values defined in params which calculate the `carriage screws`. Let me know.

ScarlettSparks replied:
OK, thanks for clarifying! In that case please use the same params from the corresponding `carriageScrews` calculations for `x` and `y` values; I know it's repetitive and verbose over there (I need to do some cleanup) so feel free to use intermediate variables here if you prefer, e.g.:

```
    xEdgeOffset = screwHeadDiam*1.5;
    yEdgeOffset = -(NEEDLE_BED_DEPTH - COMB - 6);
    xMidpoint = CAM_PLATE_WIDTH/2;
```

(Note to self: ultimately I think this calculation should probably be handled within `carriageScrews` to avoid redundancy and keep it consistent for the entire carriage, but I'll tackle that later since it might require some long-term strategy on my part.)

Neon22 replied:
ok. modified and submitted

ScarlettSparks commented:
I've got a couple of inline code questions/nitpicks but otherwise I think my next step is to try printing and using it. I don't think I'll have the opportunity to do that until next week at the earliest but I'll keep you posted! Thanks @Neon22!

Neon22 committed:
parametrized screw hole positions

using values from carriage_screws

Kyomaru1 commented:
I've printed the version of the handle in this PR and have been trying it out on the machine that I printed. 

I wasn't a fan of the solid handle aspect of this. I made a slight modification in my slicer to cut the handle in half ever so slightly so that the yarn can go into the middle cutout, but can easily be removed from the carriage without having to cut the yarn. 

I wonder if it would make sense to try taking an approach that the Bond USM had, where the handle has a large groove for the yarn to run through so that it can hang low without getting in the way of the user's hand. 

From what I've tried using my modification, it's been working well, and so long as you don't put too much pressure on the handle, I can't see a major risk of it breaking, but too much force does cause it to bend because of the break in the structure.

Neon22 commented:
@Kyomaru1  Interesting. So the Bond USM appears to control tension using the carriage directly as a yarn feeder. I think This machine is designed to use an overhead tension system.
Do you think that's right ?

I do see that that system has a set of inserts that presumably allow you to swap ~~between say intarsia and regular knitting or lacework or whatever else the system is designed to do.~~ between yarn sizes. I am not familiar with it.

Raises an interesting option - about making a modular carriage. But maybe (as we're 3D printing) maybe its just as easy to make completely different carriages.

This video has good detail. Love the transparent internals even if hidden by the handle. Would be cool to be able to see it working. Also electronic counter.
https://www.youtube.com/watch?v=5BR_1Sy2A7o

Kyomaru1 commented:
Is this machine designed for overhead tension? In my testing, as well as watching the videos posted, the yarn is just sitting on the table. Slack is pulled out manually before each row, then tension dropped as you knit across. Which is similar to how I've seen the USM used in the Cheryl Brunette videos on YouTube.

Sorry, not trying to get things off topic or anything, just raising some considerations for the design that might be beneficial to explore. Otherwise it's been good to me so far. 🙂

Neon22 commented:
@Kyomaru1 I added new PR with slotted handle - some adj possible in customiser, if you're using that in openscad - so you can set width of slot.

ScarlettSparks commented:
>  think This machine is designed to use an overhead tension system.

Not currently (although I plan to test this in the future). As @Kyomaru1 points out, this is shown pretty clearly in the videos and described in the written documentation.

> Raises an interesting option - about making a modular carriage. But maybe (as we're 3D printing) maybe its just as easy to make completely different carriages.

An earlier version of my carriage (Mk 1) used interchangeable cam segments rather than continuously adjustable cams, not exactly the same as the Bond keyplate system but kind of a similar concept. This is discussed in the design video and some further details can be found in the commit history.

@Neon22 it sounds like your proposed contributions to the project might be missing some pretty important context, both about how this machine is meant to be used and about how I've approached the design so far. While I don't necessarily expect everyone submitting an Issue or PR to know "deep lore" from early commits, I do have to expect prospective contributors to have a solid, holistic understanding of the current state of the system before trying to change or add to it. 

I see that I am responsible for this misunderstanding by not yet having explicit contributor guidelines and templates available (obviously you can't read my mind!) so I'm going to ask that you please hold off on any further work on any part of this project until I have added those and you've had a chance to read them. I do not plan to review PRs in the meantime.

As far as the handle:

Yes, I agree that a closed loop is not ideal for yarn management with the current system. Trying to avoid this is a big part of why I've been dragging my feet on getting a handle incorporated, myself. (I had written up an issue weighing the pros and cons of different handle designs, which got tragically swallowed by a browser crash, and which I really need to rewrite.) In the interim, I suspect some knitters might appreciate having anywhere at all to put their hand, so I'm open to offering a basic handle with caveats if it otherwise improves overall usability.

Neon22 commented:
Sorry Scarlett, I will withdraw.
I was so happy to see your project and I have screwed it up.

two notes before I go:
- I raised the tension arm issue with a question mark because it was a prompt for you to indicate what your plans are for tension. Not because I hadn't watched your videos, or read everything you have done here.
- "Architecture is Destiny": The shape of the problem defines how it will work.
   - please consider adding the Discussion tab to your github repo
   - there you can communicate your intentions and discuss with potential contributors issues and PRs
 
Best wishes for the future
